pkg/ddc/juicefs/transform.go

```diff
@@ -195,16 +196,20 @@ func (j *JuiceFSEngine) allocatePorts(dataset *datav1alpha1.Dataset, runtime *da
 		return err
 	}
 
-	allocatedPorts, err := allocator.GetAvailablePorts(expectedPortNum)
+	allocatedPorts, err := allocator.GetAvailablePorts(exportWorkerPodNum + exportFusePodNum)
 	if err != nil {
-		j.Log.Error(err, "can't get available ports", "expected port num", expectedPortNum)
+		j.Log.Error(err, "can't get available ports", "expected port num", exportWorkerPodNum+exportFusePodNum)
 		return err
 	}
 
 	index := 0
-	value.Worker.MetricsPort = &allocatedPorts[index]
+	if exportWorkerPodNum > 0 {
+		value.Worker.MetricsPort = &allocatedPorts[index]
+	}
 	index++
```

Comment:
Should not `index++` if exportWorkerPodNum==0.

Reply:
thks. done.